src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -106,36 +106,34 @@ private protected AssemblyLoadContext(bool representsTPALoadContext, bool isColl
             }
         }
 
+        private void RaiseUnloadEvent()
+        {
+            // Ensure that we raise the Unload event only once
```

Comment:
Note that there are subtle races where the unload event may not be delivered. Consider:

1. Unloadable context gets here, swaps in null, and gets rescheduled
2. Actual process Exit happens, delivers the event to all contexts (missing the one above) and process exits

Reply:
(Not sure what to do about it.)

Reply:
That's intended behavior - or do you think we should send the unload event twice in such case?

Reply:
I am not sure whether sending the event twice would be an improvement. I think we can leave it the way it is now.